synapse/lib/provenance.py

```diff
@@ -68,8 +68,10 @@ def claim(typ, **info):
     '''
     stack = s_task.varget('provstack')
 
-    if len(stack) > 256:  # pragma: no cover
-        raise s_exc.RecursionLimitHit(mesg='Hit global recursion limit')
+    if len(stack) > 256:
+        baseframe = stack.provs[1]
```

Comment:
Probably you'll want the last several frames. `stack.provs[-5:]` or the like.

Reply:
we probably want both of them, actually.